activesupport/lib/active_support/broadcast_logger.rb

```diff
@@ -218,6 +218,29 @@ def fatal!
       dispatch { |logger| logger.fatal! }
     end
 
+    # Adjust the log level for all loggers for the duration of the block, then restore it
+    #
+    #   logger.with_level(:debug) do
+    #     logger.debug { "Hello" }
+    #   end
+    def with_level(level, &block)
+      # Rather than just storing an array of levels and restoring based on index, store each
+      # logger with its level, in case @broadcasts is mutated inside the block and the indices
+      # no longer line up
+      previous_levels = @broadcasts.map do |logger|
+        prev_level = logger.level
+        logger.level = level
+        [logger, prev_level]
+      end
+
+      yield
+
+      # ::Logger.with_level returns `nil`, so we should return `nil`
+      nil
```

Comment:
According to the [Ruby 3.4 source](https://github.com/ruby/ruby/blob/53df2a3c548a258b383dfd026a2ba3b53ae9b9dd/lib/logger.rb#L403-L419), it appears that `Logger#with_logger` returns the value of the expression yielded by the block:

```ruby
  def with_level(severity)
    prev, level_override[Fiber.current] = level, Severity.coerce(severity)
    begin
      yield
    ensure
      if prev
        level_override[Fiber.current] = prev
      else
        level_override.delete(Fiber.current)
      end
    end
  end
```

Reply:
I probably tested this with a block that contained a single `puts` and wrongly concluded the return value was always `nil`, I can drop the explicit return value and return whatever the block returns.

Reply:
@seanpdoyle Fixed.